app/react/I18N/components/I18NLink.tsx

```diff
@@ -1,50 +1,73 @@
 /** @format */
 
-import PropTypes, { InferProps } from 'prop-types';
+import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import { Link } from 'react-router';
+import { Link, browserHistory } from 'react-router';
 import objectWithoutKeys from 'app/utils/objectWithoutKeys';
 
-const I18NLinkPropTypes = {
-  disabled: PropTypes.bool,
-  onClick: PropTypes.func,
-  children: PropTypes.any,
+const defaultProps = {
+  disabled: false,
+  onClick: (_e: any) => {},
+  confirmTitle: '',
+  confirmMessage: '',
 };
 
-type I18NLinkProps = InferProps<typeof I18NLinkPropTypes>;
+export type I18NLinkProps = typeof defaultProps & {
+  to: string;
+};
 
 export class I18NLink extends Component<I18NLinkProps> {
-  static defaultProps: I18NLinkProps;
+  static defaultProps = defaultProps;
+
+  static contextTypes = {
+    confirm: PropTypes.func,
+  };
+
+  static navigate(to: string) {
+    browserHistory.push(to);
+  }
 
   constructor(props: I18NLinkProps) {
     super(props);
     this.onClick = this.onClick.bind(this);
   }
 
   onClick(e: { preventDefault: () => void }) {
-    const { disabled, onClick } = this.props;
+    const { to, disabled, onClick, confirmTitle, confirmMessage } = this.props;
+    e.preventDefault();
     if (disabled) {
-      e.preventDefault();
       return;
     }
 
     if (onClick) {
-      onClick(e);
+      if (confirmTitle) {
+        this.context.confirm({
+          accept: () => {
+            onClick(e);
+            I18NLink.navigate(to);
+          },
+          title: confirmTitle,
+          message: confirmMessage,
+        });
+      } else {
+        onClick(e);
+        I18NLink.navigate(to);
+      }
```

Comment:
@bdittes @rustedgrail i think we should not add more functionality to i18NLink, what a bout an encapsulation of the Confirm functionality, i think this is much needed actually not only for this but we have this context.confirm thing all around the app, [this](https://itnext.io/add-confirmation-dialog-to-react-events-f50a40d9a30d) is a good option to encapsulate it on a react component, but maybe for this particular case its not clear how to use it, what do you think ?

Reply:
Hi! David and I both looked at it and concluded that it's a very elaborate solution (which I don't even fully understand since it has to intercept custom event handlers...). Also, the changes to I18NLink are not massive, so I would prefer to merge as-is.